modules/katello_repository_set.py

```diff
@@ -0,0 +1,188 @@
+        required: true
+    verify_ssl:
+        description:
+            - Verify SSL of the Foreman server
+        default: false
+    name:
+        description:
+            - Name of the repository set
+        required: true
+    product:
+        description:
+            - Name of the parent product
+        required: true
+    repositories:
+        description:
+            - Release version and base architecture of the repositories to enable
+    organization:
+        description:
+            - Organization name that the lifecycle environment is in
+        required: true
+    state:
+        description:
+            - Whether the repositories are enabled or not
+        required: true
+        choices:
+            - 'enabled'
+            - 'disabled'
+'''
+
+EXAMPLES = '''
+- name: "Enable RHEL 7 RPMs repositories"
+  katello_lifecycle_environment:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://katello.example.com"
+    verify_ssl: false
+    name: "Red Hat Enterprise Linux 7 Server (RPMs)"
+    organization: "Default Organization"
+    product: "Red Hat Enterprise Linux Server"
+    repositories:
+    - releasever: "7.0"
+      basearch: "x86_64"
+    - releasever: "7.1"
+      basearch: "x86_64"
+    - releasever: "7.2"
+      basearch: "x86_64"
+    - releasever: "7.3"
+      basearch: "x86_64"
+    state: enabled
+'''
+
+RETURN = '''# '''
+
+try:
+    from ansible.module_utils.ansible_nailgun_cement import (
+        create_server,
+        find_organization,
+        find_product,
+        find_repository_set,
+        ping_server,
+    )
+    HAS_NAILGUN_PACKAGE = True
+except:
+    HAS_NAILGUN_PACKAGE = False
+
+
+def get_desired_repos(desired_substitutions, available_repos):
+    desired_repos = []
+    for sub in desired_substitutions:
+        desired_repos += filter(lambda available: available['substitutions'] == sub, available_repos)
+    return desired_repos
+
+
+def repository_set(module, name, organization, product, state, repositories=[]):
+    changed = False
+    organization = find_organization(module, organization)
+    product = find_product(module, product, organization)
+    repo_set = find_repository_set(module, name=name, product=product)
+
+    available_repos = repo_set.available_repositories()['results']
+    current_repos = map(lambda r: r.read(), repo_set.repositories)
+    desired_repos = get_desired_repos(repositories, available_repos)
+
+    available_repo_names = set(map(lambda r: r['repo_name'], available_repos))
+    current_repo_names = set(map(lambda r: r.name, current_repos))
+    desired_repo_names = set(map(lambda r: r['repo_name'], desired_repos))
+
+    if len(desired_repo_names - available_repo_names) > 0:
+        module.fail_json(msg="Desired repositories are not available on the repository set {}. Desired: {} Available: {}"
+                         .format(name, desired_repo_names, available_repo_names))
+
+    if state == 'enabled':
```

Comment:
Now enforced via the [argument_spec](https://github.com/theforeman/foreman-ansible-modules/pull/22/files#diff-6dc7d53a8bcd06dce83d70fc7a113ab1R158)